src/ExpressionParser.php

```diff
@@ -494,9 +496,18 @@ public function getFunctionNode($name, $line)
                 }
 
                 $args = $this->parseArguments(true);
-                $class = $this->getFunctionNodeClass($name, $line);
+                $function = $this->getFunction($name, $line);
 
-                return new $class($name, $args, $line);
+                $ready = true;
+                if (!isset($this->readyNodes[$class = $function->getNodeClass()])) {
+                    $this->readyNodes[$class] = (bool) (new \ReflectionClass($class))->getConstructor()->getAttributes(FutureReady::class);
+                }
+
+                if (!$ready = $this->readyNodes[$class]) {
+                    trigger_deprecation('twig/twig', '3.12', 'Twig node "%s" is not marked as ready for passing a "TwigFunction" instead instead of its name; please make it ready and then flag it with the #[FutureReady] attribute.', $class);
```

Comment:
```suggestion
                    trigger_deprecation('twig/twig', '3.12', 'Twig node "%s" is not marked as ready for passing a "TwigFunction" instead instead of its name; please make it ready and then flag its constructor with the #[FutureReady] attribute.', $class);
```